src/cascadia/TerminalSettingsModel/ActionAndArgs.cpp

```diff
@@ -450,6 +450,36 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         return found != GeneratedActionNames.end() ? found->second : L"";
     }
 
+    // Function Description:
+    // - This will generate an ID for this ActionAndArgs, based on the ShortcutAction and the Args
+    // - It will always create the same ID if the ShortcutAction and the Args are the same
+    // - Note: this should only be called for User-created actions
+    // - Example: The "SendInput 'abc'" action will have the generated ID "User.sendInput.<hash of 'abc'>"
+    // Return Value:
+    // - The ID, based on the ShortcutAction and the Args
+    winrt::hstring ActionAndArgs::GenerateID() const
+    {
+        if (_Action != ShortcutAction::Invalid)
+        {
+            auto actionKeyString = ActionToStringMap.find(_Action)->second;
+            auto result = fmt::format(L"User.{}", std::wstring{ actionKeyString.begin(), actionKeyString.end() });
```

Comment:
You don't need to cast the `string_view` to a `string` first. Just pass `actionKeyString` directly.
For `fmt` you should always use `FMT_COMPILE`. Same thing below.

Reply:
Managed to update this one with `FMT_COMPILE`, but for the one below I think our version of `fmt` might not have the latest fixes regarding `FMT_COMPILE` with specifiers: [fix 1](https://github.com/fmtlib/fmt/issues/2719) [fix 2](https://github.com/fmtlib/fmt/issues/3586); I get compiler errors

Reply:
Ignore that, I just had to remove the `0` from `{0:X}`